packages/foundation/src/middleware/delay.ts

```diff
@@ -0,0 +1,61 @@
+import type { Request, Response, NextFunction } from "express";
+
+export function delayMiddleware(
+  delayResponses?: number | { minimum: number; maximum: number }
+) {
+  const delayMin =
+    typeof delayResponses === "number"
+      ? delayResponses
+      : delayResponses?.minimum;
+  const delayMax =
+    typeof delayResponses === "number" ? undefined : delayResponses?.maximum;
+
+  const minFromProcess = process.env.SIM_RESPONSE_DELAY_MIN
+    ? process.env.SIM_RESPONSE_DELAY_MIN
+    : process.env.SIM_RESPONSE_DELAY;
+  const min = minFromProcess ? parseInt(minFromProcess, 10) : delayMin;
+
+  const maxFromProcess = process.env.SIM_RESPONSE_DELAY_MAX;
+  const max = maxFromProcess ? parseInt(maxFromProcess, 10) : delayMax;
```

Comment:
These environment variables shouldn't be present inside a middleware, but instead should be passed in as configuration options. Environment variables can then be read and and inputs assigned at the point of entry (usually in main, or in a config module that is instantiated from main)

Reply:
Aye, we do accept those inputs and I imagine that is going to be the main code path used. However, I think this specific instance might make sense to also support this. It is, in a way, a similar case to something like the `DEBUG=*` env var and `debug` crate.

For example, if I want to quickly test a "throttle" or slow connection type situation, I can use that env var and it works for _everything_. I am not hard set in supporting this directly, but it feels worthwhile.

Reply:
We spoke and the overriding reason to remove the `process.env` is it will "win" and override any configuration passed in when (eventually) configured in a CLI. We want the CLI command args to take precedence, then `process.env`, then configuration.